libbeat/processors/ratelimit/rate_limit.go

```diff
@@ -120,6 +126,13 @@ func (p *rateLimit) makeKey(event *beat.Event) (uint64, error) {
 	return hashstructure.Hash(values, nil)
 }
 
+func (p *rateLimit) tagEvent(event *beat.Event) {
+	if p.config.MetricField != "" && p.numRateLimited.Load() > 0 {
+		event.PutValue(p.config.MetricField, p.numRateLimited.Load())
+		p.numRateLimited.Store(0)
```

Comment:
If multiple goroutines are sending events at the same time, all of them could see that `p.numRateLimited.Load() > 0`, and they could create multiple events with the same count before it is reset to 0. We could use the atomic swap to ensure that the counter is only set in one event.
```suggestion
	if p.config.MetricField != "" {
		if count := p.numRateLimited.Swap(0); count > 0 {
			event.PutValue(p.config.MetricField, count)
```